### PR TITLE
Dynamically set Auth0 SDK strategy based on request type

### PR DESCRIPTION
**Pull request by marijnvanderhorst**
## Proposed changes
Updates Auth0 SDK to the latest version and implements a middleware and event listener to dynamically set the SDK strategy based on the type of the request (`webapp` for stateful requests and `api` for stateless requests).

Fixes [AB#421](https://dev.azure.com/jdokter/973dd796-5b61-4e4a-a52b-ab66f88cef8a/_workitems/edit/421).

## Types of changes

What types of changes does your code introduce to this repository?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Unit testing
n/a

### Manual testing
Tested it manually in dataset service. @EmielSchmeink please test in user tool as well. This version can be installed using
```bash
composer require marketredesign/mrd-auth0-laravel:dev-feature/auth0-strategy
```

## Further comments
Almost all tests were broken and had to be fixed :/

### CODE REVIEW

**Review by EmielSchmeink — COMMENTED** [review]
Looks good. One small thing, and one question: the assumption is that if all tests pass, then the strategy setting works, since it didn't interfere with any of the tests?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -26,7 +26,8 @@ steps:
       sudo add-apt-repository ppa:ondrej/php
       sudo apt-get update
       sudo apt-get install php$(php_version)
-      sudo apt-get install php$(php_version)-json php$(php_version)-mbstring php$(php_version)-xml php$(php_version)-curl
+      sudo apt-get install php$(php_version)-cli php$(php_version)-mbstring php$(php_version)-xml php$(php_version)-curl
+      sudo apt-get install php$(php_version)-dom php$(php_version)-curl php$(php_version)-simplexml
```
[the remark this answers is not in the document]
Comment (marijnvanderhorst) [reply azure-pipelines.yml]: Ah indeed, good one, fixed

**Comment by marijnvanderhorst** [comment]
> the assumption is that if all tests pass, then the strategy setting works, since it didn't interfere with any of the tests?

@EmielSchmeink I don't think we can make that assumption. Added tests that verify the intended behavior now.

**Review by EmielSchmeink — APPROVED** [review]
Tests look good! Also tested with user tool, and works.

**Pull request merged**